ch-covidcertificate-backend-verifier/ch-covidcertificate-backend-verifier-ws/src/main/java/ch/admin/bag/covidcertificate/backend/verifier/ws/controller/VerificationRulesControllerV2.java

```diff
@@ -0,0 +1,97 @@
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.context.request.WebRequest;
+
+@Controller
+@RequestMapping("trust/v2")
+public class VerificationRulesControllerV2 {
+    private static final Logger logger = LoggerFactory.getLogger(VerificationRulesController.class);
+    private static String VALUE_SETS_KEY = "valueSets";
+
+    private final Map verificationRules;
+    private final String verificationRulesEtag;
+    private final ValueSetDataService valueSetDataService;
+
+    public VerificationRulesControllerV2(ValueSetDataService valueSetDataService)
+            throws IOException, NoSuchAlgorithmException {
+        ObjectMapper mapper = new ObjectMapper();
+        InputStream verificationRulesFile =
+                new ClassPathResource("verificationRulesV2.json").getInputStream();
+        this.verificationRules = mapper.readValue(verificationRulesFile, Map.class);
+        this.verificationRulesEtag =
+                EtagUtil.getSha1HashForFiles("classpath:verificationRulesV2.json");
+        this.valueSetDataService = valueSetDataService;
+    }
+
+    @Documentation(
+            description = "get list of verification rules (uses the new format)",
+            responses = {
+                "200 => list of verification rules",
+                "304 => no changes since last request"
+            },
+            responseHeaders = {"ETag:etag to set for next request:string"})
+    @GetMapping(value = "/verificationRules")
+    public @ResponseBody ResponseEntity<Map> getVerificationRules(WebRequest request)
+            throws NoSuchAlgorithmException {
+        var allIds = valueSetDataService.findAllValueSetIds();
+        HashMap<String, ArrayList<String>> valueSets = new HashMap<>();
+        ArrayList<String> strings = new ArrayList<>();
+        ObjectMapper mapper = new ObjectMapper();
+        for (var id : allIds) {
+            var valueSet = valueSetDataService.findLatestValueSet(id);
+            if (valueSet != null) {
+                try {
+                    var entryJson = mapper.readTree(valueSet);
+                    strings.add(valueSet);
+                    var fieldNameIterator = entryJson.get("valueSetValues").fieldNames();
+                    var valueSetValues = new ArrayList<String>();
+                    while (fieldNameIterator.hasNext()) {
+                        valueSetValues.add(fieldNameIterator.next());
+                    }
+                    valueSets.put(id, valueSetValues);
+                } catch (Exception ex) {
+                    continue;
+                }
+            }
+        }
+        var etag = EtagUtil.getSha1HashForStrings(strings.toArray(new String[0]));
+        if (request.checkNotModified(verificationRulesEtag + etag)) {
```

Comment:
springboot not modified magic only allows gzip compression if weak etags are used. check if appending two weak etags is still considered a valid weak etag (`W/\"etag1\"W/\"etag2\"`)